src/Evolution/BoundaryConditions/Type.hpp

```diff
@@ -0,0 +1,41 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <iosfwd>
+
+namespace evolution::boundary_conditions {
+/*!
+ * \brief The type of boundary condition.
+ *
+ * There are generally two categories or types of boundary conditions. Those
+ * that:
+ *
+ * 1. set up additional cells or elements outside the computational domain
+ *    and then apply boundary corrections the same way that is done in the
+ *    the interior. For example, in a discontinuous Galerkin scheme these
+ *    would impose boundary conditions through a numerical flux boundary
+ *    correction term.
+ *
+ * 2. change the time derivatives at the external boundary
+ */
+enum class Type {
+  /// Impose boundary conditions by setting values on ghost cells/elements.
+  ///
+  /// The ghost cell values can come either from internal data or from an
+  /// analytic prescription
+  Ghost,
+  /// Impose boundary conditions on the time derivative of the evolved
+  /// variables.
+  ///
+  /// These are imposed after all internal and external boundaries using a ghost
+  /// cell prescription have been handled.
+  TimeDerivative,
+  /// Impose ghost boundary conditions on some of the evolved variables and time
+  /// derivative boundary conditions on others.
+  GhostAndTimeDerivative
```

Comment:
If you need to support this "mixed" case, what do you need the enum for? Without having seen the code that uses this enum, it appears to me you could always assume this "mixed" case and apply whichever logic you use to decide the type of boundary condition for each variable. Or, do you want to use this enum to do compile-time optimization for the case where you _always_ apply one or the other type of boundary conditions?

Reply:
The combined case is rather complicated because when you apply the boundary condition in the algorithm is different for ghost-cell and time derivative boundary conditions.